sync-core/src/test/java/com/cloudant/sync/query/QueryExecutorMatcherTest.java

```diff
@@ -0,0 +1,39 @@
+//  Copyright (c) 2014 Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package com.cloudant.sync.query;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import com.cloudant.sync.util.SQLDatabaseTestUtils;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class QueryExecutorMatcherTest extends AbstractQueryTestBase {
```

Comment:
I think it's worth adding a comment to each of the four classes `MockSQLOnlyQueryExecutor`, `MockSQLOnlyIndexManager`, `MockMatcherQueryExecutor` and `MockMatcherIndexManager` to explain what each one does -- or you could point to explanations you add to `QueryExecutorMatcherTest` and `QueryExecutorSQLOnlyTest`.

This set up is rather odd -- perhaps we can figure out a way to clean up both the objc and this one sometime! -- and so needs documenting.